File: android-lib/src/androidTest/java/com/kinvey/androidTest/model/PersonArray.java
package com.kinvey.androidTest.model;

import com.google.api.client.util.Key;

import java.util.List;

public class PersonArray extends Person {

    public static final String COLLECTION = "PersonArray";

    @Key
    private PersonArray[] array;

    @Key
    private PersonArray personArray;

    public PersonArray() {
    }

    public PersonArray(String name) {
        this.username = name;
    }

    public PersonArray[] getArray() {
        return array;
    }

    public void setArray(PersonArray[] array) {
        this.array = array;
    }

    public PersonArray getPersonArray() {
        return personArray;
    }

    public void setPersonArray(PersonArray personArray) {
        this.personArray = personArray;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/model/PersonArray.kt
package com.kinvey.androidTest.model

import com.google.api.client.util.Key

class PersonArray : Person {
    @Key
    var array: Array<PersonArray>
    @Key
    var personArray: PersonArray? = null

    constructor() {}
    constructor(name: String?) {
        username = name
    }

    companion object {
        const val COLLECTION = "PersonArray"
    }
}